### ISSUE TITLE
404 when hosting using `-fs` and hosting over `net/http`

### ISSUE THREAD

**Issue by clstrco** [go-bindata/go-bindata#62]
I tried using the `-fs` flag on this project by following the instructions [at the bottom of the readme](https://github.com/go-bindata/go-bindata#serve-assets-with-nethttp) but the http routes are all 404 Not Found.

Here's a repo that reproduces the pathology, including go tests:

https://github.com/clstrco/gbdfail

I had to split the logic up for registering the routes from main so it could be used in the tests; that's found [here](https://github.com/clstrco/gbdfail/blob/master/gbdfail.go). The test code can be found [here](https://github.com/clstrco/gbdfail/blob/master/gbdfail_test.go).

The project pins `go-bindata` at latest release found by the go tool (see the [go.mod file](https://github.com/clstrco/gbdfail/blob/master/go.mod))

Please let me know if I can provide other info.

**Comment by tooolbox** [comment]
I just looked at your sample repo.

Your problem is that you're stripping the prefix when you're generating the bindata files.  So the files are all available at `/a/...` and `/b/...` and so on.  If you still want them to be at `/s/a/...` then don't use the prefix-stripping flag in the bindata command.  Your `http.StripPrefix` is basically "putting it back".

If your question is answered, I think the issue can be closed.

**Comment by fdully** [comment]
I've got the same issue. I agree that docs must be changed about -fs key.
If I do like in docs than it is not working.

If I do like this
`/go/bin/go-bindata -o static.go -fs static/`
and it is working only like this
`mux.Handle("/static/", http.FileServer(AssetFile()))`

if I do like this
`/go/bin/go-bindata -o static.go -fs -prefix "static/" static/`
and it is working only like this
`mux.Handle("/static", http.StripPrefix("/static", http.FileServer(AssetFile())))`

**Comment by darkermatter** [comment]
@fdully The docs are correct. Have you read the documentation for ```http.StripPrefix()```? If you did, I believe you misunderstood it. Possibly, you are confusing routes with directories.

Your first example proves the docs right.
The second one does too... You are stripping the prefix to match what you have in the bindata which has no prefix ```static/```

**Comment by fdully** [comment]
@darkermatter Yes, docs are correct. I agree. I have confused routes with directories. 

I'am just saying that is better to add few words to it. 

something like "-fs path to filename will be equal to http path in url. for `-fs static/` http path must be /static/filename, for `-fs -prefix "static/" static/` http path must be /filename"
